app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/repository/CafeteriaRemoteRepository.kt

```diff
@@ -29,8 +30,7 @@ class CafeteriaRemoteRepository @Inject constructor(
                 .doOnNext { localRepository.clear() }
                 .flatMap { tumCabeClient.cafeterias }
                 .doAfterNext { localRepository.updateLastSync() }
-                .map { it.toTypedArray() }
-                .subscribe(localRepository::addCafeteria, Utils::log)
+                .subscribe(Consumer { localRepository.addCafeteria(it) }, Consumer { Utils.log(it) })
```

Comment:
You can shorten this to: `subscribe(localRepository::addCafeteria, Utils::log)`.